### PR TITLE
Adding cni repair functionality to the istioctl installer.

### PR DESCRIPTION
**Pull request by stewartbutler**
This adds the CNI repair container to the cni installation via istioctl. It also fixes the makefiles so that `make gen` works again, though I'm not sure it does it correctly.

### CODE REVIEW

File: operator/Makefile [hunk 1]
```diff
@@ -112,11 +112,11 @@ RUN = $(CONTAINER_CLI) run -t -i --sig-proxy=true -u $(UID):$(GID) --rm \
 	-v /etc/passwd:/etc/passwd:ro \
 	$(DOCKER_SOCKET_MOUNT) \
 	$(CONTAINER_OPTIONS) \
-	--mount type=bind,source="$(PWD)",destination="/work" \
+	--mount type=bind,source="$(PWD)/../",destination="/work" \
```
Comment (howardjohn) [inline operator/Makefile]: this file shouldn't be used and the licenses files under operator shouldn't be checked in (they are duplicates of too level). We should not be calling make within make, just including other makefiles

**Comment by stewartbutler** [comment]
@howardjohn I think this should be cherrypicked into 1.5. However, it requires a `make gen` to regenerate the protos, so do I need a separate PR for it?

File: operator/Makefile (hunk 1 above)
Comment (stewartbutler) [reply operator/Makefile]: Fixed

File: go.sum [hunk 2]
```diff
@@ -453,6 +453,7 @@ github.com/lestrrat-go/jwx v0.9.0/go.mod h1:iEoxlYfZjvoGpuWwxUz+eR5e6KTJGsaRcy/Y
 github.com/liggitt/tabwriter v0.0.0-20181228230101-89fcab3d43de h1:9TO3cAIGXtEhnIaL+V+BEER86oLrvS+kWobKpbJuye0=
 github.com/liggitt/tabwriter v0.0.0-20181228230101-89fcab3d43de/go.mod h1:zAbeS9B/r2mtpb6U+EI2rYA5OAXxsYw6wTamcNW+zcE=
 github.com/lithammer/dedent v1.1.0/go.mod h1:jrXYCQtgg0nJiN+StA2KgR7w6CiQNv9Fd/Z9BP0jIOc=
+github.com/lyft/protoc-gen-validate v0.0.13 h1:KNt/RhmQTOLr7Aj8PsJ7mTronaFyx80mRTT9qF261dA=
```
Comment (howardjohn) [inline go.sum]: I suspect this change should not be here

File: manifests/istio-cni/templates/daemonset.yaml [hunk 3]
```diff
@@ -78,6 +78,36 @@ spec:
               name: cni-bin-dir
             - mountPath: /host/etc/cni/net.d
               name: cni-net-dir
+        - name: repair-cni
+            {{- if contains "/" .Values.cni.image }}
+          image: "{{ .Values.cni.image }}"
+            {{- else }}
+          image: "{{ .Values.cni.hub | default .Values.global.hub }}/{{ .Values.cni.image | default "install-cni" }}:{{ .Values.cni.tag | default .Values.global.tag }}"
+            {{- end }}
+          imagePullPolicy: {{ .Values.cni.pullPolicy | default .Values.global.imagePullPolicy }}
```
Comment (howardjohn) [inline manifests/istio-cni/templates/daemonset.yaml]: nit: this basically reverts https://github.com/istio/istio/pull/20707

File: manifests/istio-cni/templates/daemonset.yaml [hunk 4]
```diff
@@ -78,6 +78,36 @@ spec:
               name: cni-bin-dir
             - mountPath: /host/etc/cni/net.d
               name: cni-net-dir
+        - name: repair-cni
```
Comment (howardjohn) [inline manifests/istio-cni/templates/daemonset.yaml]: is this supposed to be optional

**Review by howardjohn — COMMENTED** [review]
generally lgtm

File: go.sum (hunk 2 above)
Comment (stewartbutler) [reply go.sum]: Removed

File: manifests/istio-cni/templates/daemonset.yaml (hunk 4 above)
Comment (stewartbutler) [reply manifests/istio-cni/templates/daemonset.yaml]: Now optional.

File: manifests/istio-cni/templates/daemonset.yaml (hunk 3 above)
Comment (stewartbutler) [reply manifests/istio-cni/templates/daemonset.yaml]: Added the wrapper from #20707

**Review by costinm — APPROVED** [review]

**Pull request merged**